src/main/java/org/folio/helper/CheckinReceivePiecesHelper.java

```diff
@@ -518,23 +525,22 @@ protected Future<ReceiptStatus> calculatePoLineReceiptStatus(PoLine poLine, List
    */
   private Future<ReceiptStatus> calculatePoLineReceiptStatus(int expectedPiecesQuantity, PoLine poLine,
       List<Piece> pieces, RequestContext requestContext) {
-    // Fully Received:If receiving and there is no expected piece remaining
-    if (!poLine.getCheckinItems().equals(Boolean.TRUE) && expectedPiecesQuantity == 0) {
```

Comment:
sonar requires previous version of  construction: 
!poLine.getCheckinItems().equals(Boolean.TRUE)